### PR TITLE
Enable cross-repo navigation.

### LINKED ISSUE #150: Exception indexing google/guava repo from parsing descriptor from a local symbol
```
$ lsif-java index
...
        Generating LSIF... [          ] 0%   8 files processedjava.lang.IllegalArgumentException: invalid symbol format
[error] local4
[error]      ^
[error] 	at com.sourcegraph.lsif_semanticdb.SymbolDescriptor$Parser.fail(SymbolDescriptor.java:46)
[error] 	at com.sourcegraph.lsif_semanticdb.SymbolDescriptor$Parser.parseDescriptor(SymbolDescriptor.java:117)
[error] 	at com.sourcegraph.lsif_semanticdb.SymbolDescriptor$Parser.entryPoint(SymbolDescriptor.java:34)
[error] 	at com.sourcegraph.lsif_semanticdb.SymbolDescriptor.parseFromSymbol(SymbolDescriptor.java:16)
[error] 	at com.sourcegraph.lsif_semanticdb.SignatureFormatter.symbolDisplayName(SignatureFormatter.java:363)
[error] 	at com.sourcegraph.lsif_semanticdb.SignatureFormatter.formatType(SignatureFormatter.java:245)
```

For local symbols, you need to look up in the symbol table to get the descriptor name.

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, lsif-java was only able to index a single repositories. It
was not possible to navigate between repositories. Now, the emitted LSIF
contains `packageInformation` nodes that document how to link LSIF
indexes between repositories.

Rough summary of the changes in this commit:

- New `LsifBuildTool` that we use to index `*-sources.jar` of
  Java libraries and the JDK sources.
- New `ClasspathEntry` class to represent a Maven library. We extract
  `ClasspathEntry` values from the build via text files in the
  SemanticDB targetroot.
- Miscellaneous small bugfixes while indexing the JDK and Guava.
- Fixes #150.
- Improved error reporting to debug issues in the `javac` wrapper that
  we use for Gradle and Maven.

### CODE REVIEW

File: lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala [hunk 1]
```diff
@@ -92,6 +100,38 @@ class GradleBuildTool(index: IndexCommand) extends BuildTool("Gradle", index) {
...
+          |          configurations.each { config ->
+          |              def artifactType = org.gradle.api.attributes.Attribute.of("artifactType", String.class)
+          |              def attributeType = "jar"
+          |              if (config.canBeResolved) {
+          |                def artifacts = config.incoming.artifactView { view ->
+          |                  view.lenient = true
+          |                  view.attributes { container ->
+          |                    container.attribute(artifactType, attributeType)
+          |                  }
+          |                }.artifacts
```
Comment (Strum355) [inline lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala:117]: Really interesting! How did you come across this solution?

Comment (olafurpg) [reply lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala:117]: I adapted from the Bloop plugin https://github.com/scalacenter/bloop/blob/4d9f966b67ff726855cb36ec3565a34d00bf41ec/integrations/gradle-bloop/src/main/scala/bloop/integrations/gradle/model/BloopConverter.scala#L277-L299

Comment (Strum355) [reply lsif-java/src/main/scala/com/sourcegraph/lsif_java/buildtools/GradleBuildTool.scala:117]: Damn, wish I knew of this many months ago :laughing:

File: semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbVisitor.java [hunk 2]
```diff
@@ -74,6 +74,7 @@ public SemanticdbVisitor(
 
   private <T extends JCTree & JCDiagnostic.DiagnosticPosition> void emitSymbolOccurrence(
       Symbol sym, T posTree, Role role, CompilerRange kind) {
+    if (sym == null) return;
```
Comment (Strum355) [inline semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbVisitor.java:77]: Is this an issue you came across?

Comment (olafurpg) [reply semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbVisitor.java:77]: Yeah, hit on three NPEs when compiling the JDK

**Review by Strum355 — APPROVED** [review]
:tada: awesome work! A lot to unpack here but LGTM :slightly_smiling_face:

**Pull request merged**